napari/utils/migrations.py

```diff
@@ -218,3 +218,71 @@ def __init_subclass__(cls, **kwargs):
     _OldClass.__new__.__signature__ = prealloc_signature  # type: ignore [attr-defined]
 
     return _OldClass
+
+
+class DeprecatingDict(dict[str, Any]):
```

Comment:
I wonder if we should use `wrapt.ObjectProxy` instead of just overriding as the implementation is similar to other things we use that for: https://github.com/napari/napari/blob/f327d1f545b88c6473f3103cc46e8beac194d1ea/napari/utils/_proxies.py#L15

Reply:
I don't have an opinion about it, but if this is working, we can just use it. It is nice and simple. Sometimes wrapt feels a little too magical 😅